libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1398,6 +1402,18 @@ class MapboxNavigation @VisibleForTesting internal constructor(
     @JvmOverloads
     fun setRerouteController(
         rerouteController: NavigationRerouteController? = defaultRerouteController
+    ) {
+        val oldController = this.rerouteController
```

Comment:
I suggest we don't copy-paste the logic but reuse the existing method with an adapter as an argument.

Reply:
thanks